### DISCUSSION TITLE
I2C.writeto mem difference between 1.18 and 1.19

### DISCUSSION THREAD

**Discussion by H-sw123** [micropython/micropython#10089] [Libraries & Drivers]
Hello everyone! I used the latest 1.19 version of firmware, which caused my program to die. After checking, I found that it died in I2C.writeto mem, there is no problem with my 1.18 version of firmware. Below is my attached picture, please help me to have a look.

![image](https://user-images.githubusercontent.com/81347510/204080755-71b3c067-712f-497f-b6eb-6dc750da0eac.png)
![image](https://user-images.githubusercontent.com/81347510/204080891-b9012868-2b68-4760-a480-509db965dc8d.png)

**Comment by robert-hh** [comment]
It would be interesting to see the error message of the board. Or is there none? Maybe run the script from a simple serial terminal, not Thonny, or if it has to be Thonny, copy the script to the board and run `import scriptname` from the REPL command line.

**Reply by H-sw123** [reply]
Thank you for your reply. It's not the answer I needed.
 My problem is that I use 1.18 firmware and can write to my sensor normally, but when I use 1.19.1 firmware, when I run the program, it won't work.

**Reply by robert-hh** [reply]
But to help you, we have to know what happens. And Thonny is an excellent tool for bug hiding.

**Reply by robert-hh** [reply]
b.t.w.: How to you create the I2C object? To which pins is the I2C device connected. We need as well see more of your script, need to know more of your set-up..

**Comment by jimmo** [comment]
@H-sw123 I suspect @robert-hh is right, can you share how you create teh I2C object.

My guess is that you don't specify the pins (i.e. just `machine.I2C(0)`, and since v1.18 we switched to using the Raspberry Pi's official default pins which are likely different to what you're wired to). Try adding `machine.I2C(0, sda=machine.Pin(N), scl=machine.Pin(N))` to explicitly choose the pins.

**Reply by H-sw123** [reply]
Thank you for your reply. I have the designated pin.
 self.i2c = I2C(0, scl=Pin(21), sda=Pin(20), freq=100000)

**Comment by H-sw123** [comment]
Thank you for your answers! There was a problem with my sensor output, which caused I2C to die when writing commands. Now it has been solved.
